src/libcsg/modules/io/h5mdtrajectoryreader.cc

```diff
@@ -295,6 +309,26 @@ bool H5MDTrajectoryReader::NextFrame(Topology &top) {  // NOLINT const reference
   return true;
 }
 
-}  // end csg
-}  // end votca
+double* H5MDTrajectoryReader::ReadBox(hid_t ds, hid_t ds_data_type, int row) {
+  hsize_t offset[2];
+  offset[0] = row;
+  offset[1] = 0;
+  hsize_t ch_rows[2];
+  ch_rows[0] = 1;
+  ch_rows[1] = 3;
+  hid_t dsp = H5Dget_space(ds);
+  H5Sselect_hyperslab(dsp, H5S_SELECT_SET, offset, NULL, ch_rows, NULL);
+  hid_t mspace1 = H5Screate_simple(2, ch_rows, NULL);
+  double *data_out = new double[3];
```

Comment:
Is there anyway we can avoid using new and raw pointers here?

Reply:
maybe declare unique_ptr and pass data_out.get() to H5Dread, then you don't need to worry about deleting it later.